# Datasets para el estudio de personas con ataques cardíacos

In [1]:
#Importaciones
import pandas as pd

## Dataset "healthcare-dataset-stroke-data.csv"

In [2]:
#Leemos dataset descargado desde aquí: https://gist.github.com/aishwarya8615/d2107f828d3f904839cbcb7eaa85bd04#file-healthcare-dataset-stroke-data-csv
df_healthcare_stroke = pd.read_csv('datasets/healthcare-dataset-stroke-data.csv')
df_healthcare_stroke.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

Podemos observar datos básicos a nivel salud y sociales de cada persona que se entienden en su mayoría. BMI se refiere al índice al "Body mass index", el índice de masa corporal.

In [3]:
df_healthcare_stroke.info()
df_healthcare_stroke.describe().T

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


count          mean           std    min        25%  \
id                 5110.0  36517.829354  21161.721625  67.00  17741.250   
age                5110.0     43.226614     22.612647   0.08     25.000   
hypertension       5110.0      0.097456      0.296607   0.00      0.000   
heart_disease      5110.0      0.054012      0.226063   0.00      0.000   
avg_glucose_level  5110.0    106.147677     45.283560  55.12     77.245   
bmi                4909.0     28.893237      7.854067  10.30     23.500   
stroke             5110.0      0.048728      0.215320   0.00      0.000   

                         50%       75%       max  
id                 36932.000  54682.00  72940.00  
age                   45.000     61.00     82.00  
hypertension           0.000      0.00      1.00  
heart_disease          0.000      0.00      1.00  
avg_glucose_level     91.885    114.09    271.74  
bmi                   28.100     33.10     97.60  
stroke                 0.000      0.00      1.00

Observamos que hay 5110 registros, pero sólo 4909 registros tienen BMI (índice de masa corporal).

## Dataset "Behavioral_Risk_Factor_Data__Heart_Disease___Stroke_Prevention.csv"

In [4]:
#Leemos dataset descargado desde aquí: https://data.world/cdc/behavioral-risk-factor-heart
df_behavioral_risk_factor = pd.read_csv('datasets/Behavioral_Risk_Factor_Data__Heart_Disease___Stroke_Prevention.csv')
df_behavioral_risk_factor.head()

Year LocationAbbr LocationDesc Datasource PriorityArea1 PriorityArea2  \
0  2011           NV       Nevada      BRFSS          None          None   
1  2011           NV       Nevada      BRFSS          None          None   
2  2011           NV       Nevada      BRFSS          None          None   
3  2011           NV       Nevada      BRFSS          None          None   
4  2011           NV       Nevada      BRFSS          None          None   

  PriorityArea3 PriorityArea4                 Category  \
0          None          None  Cardiovascular Diseases   
1          None          None  Cardiovascular Diseases   
2          None          None  Cardiovascular Diseases   
3          None          None  Cardiovascular Diseases   
4          None          None  Cardiovascular Diseases   

                    Topic  ...  \
0  Coronary Heart Disease  ...   
1  Coronary Heart Disease  ...   
2  Coronary Heart Disease  ...   
3  Coronary Heart Disease  ...   
4  Coronary Heart Disease  ...   

                                 Data_Value_Footnote Confidence_Limit_Low  \
0  Statistically unstable estimates not presented...                  NaN   
1  Statistically unstable estimates not presented...                  NaN   
2  Statistically unstable estimates not presented...                  NaN   
3  Statistically unstable estimates not presented...                  NaN   
4  Statistically unstable estimates not presented...                  NaN   

  Confidence_Limit_High CategoryID TopicID  IndicatorID BreakoutCategoryID  \
0                   NaN         C1      T4        BR001              BOC03   
1                   NaN         C1      T4        BR001              BOC04   
2                   NaN         C1      T4        BR001              BOC03   
3                   NaN         C1      T4        BR001              BOC04   
4                   NaN         C1      T4        BR001              BOC04   

  BreakOutID  LocationID                                GeoLocation  
0      AGE04        32.0  (39.493240390000494, -117.07184056399967)  
1      RAC02        32.0  (39.493240390000494, -117.07184056399967)  
2      AGE01        32.0  (39.493240390000494, -117.07184056399967)  
3      RAC07        32.0  (39.493240390000494, -117.07184056399967)  
4      RAC04        32.0  (39.493240390000494, -117.07184056399967)  

[5 rows x 27 columns]

Se observan muchos datos (27 columnas) por individuo. Este dataset requiere más conocimiento del dominio para poder analizarlo en detalle.

In [5]:
df_behavioral_risk_factor.info()
df_behavioral_risk_factor.describe().T

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35004 entries, 0 to 35003
Data columns (total 27 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Year                        35004 non-null  int64  
 1   LocationAbbr                35004 non-null  object 
 2   LocationDesc                35004 non-null  object 
 3   Datasource                  35004 non-null  object 
 4   PriorityArea1               35004 non-null  object 
 5   PriorityArea2               35004 non-null  object 
 6   PriorityArea3               35004 non-null  object 
 7   PriorityArea4               35004 non-null  object 
 8   Category                    35004 non-null  object 
 9   Topic                       35004 non-null  object 
 10  Indicator                   35004 non-null  object 
 11  Break_Out_Category          35004 non-null  object 
 12  Break_out                   35004 non-null  object 
 13  Data_Value_Type             350

count         mean        std     min     25%  \
Year                   35004.0  2012.246600   1.088955  2011.0  2011.0   
Data_Value             31850.0    36.370628  24.805992     0.2    19.2   
Confidence_Limit_Low   31482.0    32.607573  24.060026     0.0    15.8   
Confidence_Limit_High  31482.0    40.178076  25.575972     0.3    22.5   
LocationID             35004.0    28.311479  16.037427     0.5    15.0   

                          50%     75%     max  
Year                   2012.5  2013.0  2014.0  
Data_Value               30.3    47.7   100.0  
Confidence_Limit_Low     26.7    41.5    97.3  
Confidence_Limit_High    33.8    56.4   100.0  
LocationID               28.0    41.0    56.0

Observamos 35004 tuplas, no todas con valores para campos como los footnote.

## Dataset "National_Health_Interview_Survey__NHIS__-_National_Cardiovascular_Disease_Surveillance_Data.csv"

In [6]:
#Leemos dataset descargado desde aquí: https://data.world/cdc/nhis-national-cardiovascular
df_survey = pd.read_csv('datasets/National_Health_Interview_Survey__NHIS__-_National_Cardiovascular_Disease_Surveillance_Data.csv')
df_survey.head()

Year LocationAbbr LocationDesc DataSource    PriorityArea1 PriorityArea2  \
0  2000          MDW      Midwest       NHIS  Million Hearts®          None   
1  2000          MDW      Midwest       NHIS  Million Hearts®          None   
2  2000          MDW      Midwest       NHIS  Million Hearts®          None   
3  2000          MDW      Midwest       NHIS  Million Hearts®          None   
4  2000          MDW      Midwest       NHIS  Million Hearts®          None   

  PriorityArea3 PriorityArea4                 Category  \
0          None          None  Cardiovascular Diseases   
1          None          None  Cardiovascular Diseases   
2          None          None  Cardiovascular Diseases   
3          None          None  Cardiovascular Diseases   
4          None          None  Cardiovascular Diseases   

                    Topic  ... Data_Value_Footnote Confidence_limit_Low  \
0  Coronary Heart Disease  ...                 NaN                  3.3   
1  Coronary Heart Disease  ...                 NaN                  2.2   
2  Coronary Heart Disease  ...                 NaN                  0.5   
3  Coronary Heart Disease  ...                 NaN                  3.2   
4  Coronary Heart Disease  ...                 NaN                  3.3   

  Confidence_limit_High CategoryId TopicId  IndicatorID  BreakOutCategoryId  \
0                   4.5         C1      T4        NS001               BOC04   
1                   3.4         C1      T4        NS001               BOC02   
2                   2.8         C1      T4        NS001               BOC04   
3                   5.2         C1      T4        NS001               BOC03   
4                   4.4         C1      T4        NS001               BOC01   

   BreakOutId  LocationID  GeoLocation  
0       RAC01         103          NaN  
1       GEN02         103          NaN  
2       RAC04         103          NaN  
3       AGE05         103          NaN  
4       OVR01         103          NaN  

[5 rows x 27 columns]

Dataset muy similar al anterior (diferente fuente, no obstante)

In [7]:
df_survey.info()
df_survey.describe().T

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4455 entries, 0 to 4454
Data columns (total 27 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Year                        4455 non-null   int64  
 1   LocationAbbr                4455 non-null   object 
 2   LocationDesc                4455 non-null   object 
 3   DataSource                  4455 non-null   object 
 4   PriorityArea1               4455 non-null   object 
 5   PriorityArea2               4455 non-null   object 
 6   PriorityArea3               4455 non-null   object 
 7   PriorityArea4               4455 non-null   object 
 8   Category                    4455 non-null   object 
 9   Topic                       4455 non-null   object 
 10  Indicator                   4455 non-null   object 
 11  Break_Out_Category          4455 non-null   object 
 12  Break_Out                   4455 non-null   object 
 13  Data_Value_Type             4455 

count         mean        std     min     25%  \
Year                        4455.0  2007.098765   4.348341  2000.0  2003.0   
Data_Value                  4455.0    18.821841  24.752340     0.0     2.8   
Data_Value_Footnote_Symbol     0.0          NaN        NaN     NaN     NaN   
Data_Value_Footnote            0.0          NaN        NaN     NaN     NaN   
Confidence_limit_Low        4384.0    17.613549  24.350170     0.0     2.2   
Confidence_limit_High       4384.0    20.913618  25.173631     0.1     3.8   
LocationID                  4455.0    82.000000  41.016797     0.0   101.0   
GeoLocation                    0.0          NaN        NaN     NaN     NaN   

                               50%     75%     max  
Year                        2007.0  2011.0  2014.0  
Data_Value                     6.5    25.5    98.7  
Data_Value_Footnote_Symbol     NaN     NaN     NaN  
Data_Value_Footnote            NaN     NaN     NaN  
Confidence_limit_Low           6.0    23.3    98.0  
Confidence_limit_High          9.9    28.5    99.1  
LocationID                   102.0   103.0   104.0  
GeoLocation                    NaN     NaN     NaN

Mismo número de campos que el anterior, pero con muchos menos registros. ¿Puede que sea una extracción?